src/TemplateCompiler/NodeFactory/VarDocNodeFactory.php

```diff
@@ -10,18 +10,37 @@
 
 final class VarDocNodeFactory
 {
+    /**
+     * @var array <string, Nop>
+     */
+    private static array $docNodes = [];
+
+    public function __construct()
+    {
+        self::$docNodes = [];
+    }
+
     /**
      * @param VariableAndType[] $variablesAndTypes
      * @return Nop[]
      */
     public function createDocNodes(array $variablesAndTypes): array
     {
-        $docNodes = [];
         foreach ($variablesAndTypes as $variableAndType) {
-            $docNodes[$variableAndType->getVariable()] = $this->createDocNop($variableAndType);
+            if (isset(self::$docNodes[$variableAndType->getVariable()])) {
+                // avoids overwriting the same variable, if it is preset
+                continue;
+            }
+
+            self::$docNodes[$variableAndType->getVariable()] = $this->createDocNop($variableAndType);
         }
 
-        return array_values($docNodes);
+        $values = array_values(self::$docNodes);
+
+        // reset for next run
+        self::$docNodes = [];
```

Comment:
Why not just keep the variable as a local rather then a static?

Reply:
to call it statically in ViewVariableAnalyzer, if i can resolve the same instance as is already initilized for the DocNodeFactory that would also solve it. But i cant seem to figure out the dependency injection in this.